modules/base/src/processors/noisegenerator2d.cpp

```diff
@@ -154,36 +156,35 @@ NoiseProcessor::NoiseProcessor()
     typeOnChange();
 }
 
-NoiseProcessor::~NoiseProcessor() {}
+NoiseGenerator2D::~NoiseGenerator2D() {}
 
-void NoiseProcessor::process() {
+void NoiseGenerator2D::process() {
     if (useSameSeed_.get()) {
         mt_.seed(seed_.get());
     }
 
     std::uniform_real_distribution<float> r(range_.get().x, range_.get().y);
-    std::shared_ptr<Image> img;
+    std::shared_ptr<Layer> layer;
```

Comment:
You might add a LayerInformationProperty to this so you can sett some "meta" data if needed

Reply:
also maybe the basis property